### PR TITLE
Added custom api support

### LINKED ISSUE #38: Provide custom API loading
We need a method of exposing custom APIs.  They should be loaded / imported using the same mechanism as we use for tables.

### PR DESCRIPTION
**Pull request by mamaso**
resolves #38

### CODE REVIEW

File: src/express/api/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    var router = express.Router();
+
+    /**
+    Register a single api with the specified definition.
+    @function add
+    @param {string} name - The name of the api. HTTP operations will be exposed on this route.
+    @param {apiDefinition} definition - The definition for the api
+    */
+    router.add = function (name, definition) {
+        assert(name, 'A api name was not specified');
```
Comment (phvannor) [inline src/express/api/index.js]: nit: "An API name was not specified"

File: src/express/api/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        var apiRouter = express.Router();
+
+        Object.getOwnPropertyNames(definition).forEach(function (property) {
+            if (supportedVerbs.some(function (verb) { return verb === property; })) {
```
Comment (phvannor) [inline src/express/api/index.js]: preflight requests work right?  You just can't add your own code on them?

File: src/express/api/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                if (definition.authorise || definition[property].authorise) {
+                    delete definition[property].authorise;
+                    definition[property] = [authorise].concat(definition[property]);
+                    logger.debug("Adding authorisation to " + property + " for api " + name);
+                }
+                logger.debug("Adding method " + property + " to api " + name);
```
Comment (phvannor) [inline src/express/api/index.js]: So I can only have verbs on my API name?  Or did I miss something in how I can add /calculator/add type endpoints?

File: src/express/api/index.js (hunk 2 above)
Comment (mamaso) [inline src/express/api/index.js]: Correct, just added a test for preflight behavior

File: src/express/api/index.js (hunk 1 above)
Comment (mamaso) [inline src/express/api/index.js]: fixed

File: src/express/api/index.js (hunk 3 above)
Comment (danderson00) [inline src/express/api/index.js]: Remember that this is just syntactic sugar for attaching custom APIs, and to support the portal "turnkey" experience. There is nothing to stop a user writing the extra line of code to hook up to express manually:

    app.get('/api/calculator/add', function (req, res, next) {})

Having said this, we should consider adding support for multi-level paths like this.

**Comment by danderson00** [comment]
I added a new issue for multi level paths - https://github.com/Azure/azure-mobile-apps-node/issues/96

**Pull request merged**